homeassistant/helpers/service.py

```diff
@@ -113,3 +117,58 @@ def extract_entity_ids(hass, service_call):
         return group.expand_entity_ids(hass, [service_ent_id])
 
     return [ent_id for ent_id in group.expand_entity_ids(hass, service_ent_id)]
+
+
+# pylint: disable=too-few-public-methods
+class CallRateDelayThrottle(object):
```

Comment:
@balloob, I would prefer to just make it async.  Just tell me how I can achieve that. I have to change it anyway to get it merged.